Glasgow Class 6 - Mandy Cheung - JS1 - Week 2

m4ndycheung opened the pull request:
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Mandy Cheung
- Your City: Glasgow
- Your Slack Name: Mandy Cheung

# Homework Details

- Module: JS1
- Week: 2

# Notes

- What did you find easy?
- Comparison and Logical Operators
- The syntax is getting easier with practice!

- What did you find hard?
- Telling my brain to stop thinking about the most complicated way to solve problems (especially the reverseCount)
- Testing my code works in these npm test files, asked a friend how to print it to the console

- What do you still not understand?

- Any other notes?

m4ndycheung committed:
Completed 1-fix-functions, added to 2-function-creation

m4ndycheung committed:
Completed discount, odd numbers, buy two get cheapest free

m4ndycheung committed:
Added to function creation

Added an if statement

m4ndycheung committed:
Added an else if statement to canRegister

Tested it and it works out.

m4ndycheung committed:
Completed 2-function creation

m4ndycheung committed:
Update 2-function-creation.js

msimmdev commented on mandatory/2-function-creation.js:
```diff
@@ -15,26 +21,58 @@ function isAcceptableUser(userAge, isLoggedIn) {}
+    if (count % 2 === 1) {
+    console.log(count);
+    }
+    count = count + 1;
```
Nitpick: `count++` is a shorter way of expressing `count = count + 1`

m4ndycheung replied:
Thanks, Michael, noted!

msimmdev commented on mandatory/2-function-creation.js:
```diff
@@ -15,26 +21,58 @@ function isAcceptableUser(userAge, isLoggedIn) {}
+
+  while (count < limit) {
+    if (count % 2 === 1) {
+    console.log(count);
```
Nitpick: This line should be indented by one more tab

msimmdev commented on mandatory/2-function-creation.js:
```diff
@@ -45,7 +83,12 @@ function canRegister(age) {}
 
-function countReverse(number) {}
+function countReverse(number) {
+  while (number >= 1) {
```
Suggestion: create a new variable that you can loop over and decrement rather than using the input `number`. e.g.
```
let currentNumber = number;
while (currentNumber  >= minNum) {
  console.log(number);
  currentNumber--;
}
```
The reason for this is you still have access to the unmodified user input in case you need to use it in  a future modification of the function.
Also you'll find in the future lessons that for certain types passed to a function modifying the input data could have an impact on other parts of the program (this isn't a problem for integers though so wouldn't be an issue here)

m4ndycheung replied:
Thanks, Michael, I think I need you to explain this a bit more to me over a buddy chat so I'll bring this up.

msimmdev reviewed (commented):
Praise: Well done on these questions I found no issues in the answers given, just some suggestions and nitpicks.

m4ndycheung committed:
Made changes to 2-function-creation applyDiscount answer

m4ndycheung committed:
Added answer to extra

m4ndycheung committed:
Amended answer to 2-function-creation

Took on Michael's feedback to write count = count + 1 in a shorter way

m4ndycheung committed:
Updated answers for 2-function-creation

Took on Michael's feedback and added indentation to printOddNumbers.

Also declared a let variable for countReverse so that input data won't be modified
